File: apps/patter-voice-sdk/libraries/python/getpatter/llm/google.py
"""Google Gemini LLM for Patter pipeline mode."""

from __future__ import annotations

import os
from typing import ClassVar

from getpatter.providers.google_llm import GoogleLLMProvider as _GoogleLLM

__all__ = ["LLM"]


class LLM(_GoogleLLM):
    """Google Gemini LLM provider (``google-genai`` SDK).

    Example::

        from getpatter.llm import google

        llm = google.LLM()                            # reads GEMINI_API_KEY or GOOGLE_API_KEY
        llm = google.LLM(api_key="AIza...", model="gemini-2.5-flash")
    """

    provider_key: ClassVar[str] = "google"

    def __init__(
        self,
        api_key: str | None = None,
        *,
        model: str = "gemini-2.5-flash",
        **kwargs,
    ) -> None:
        # Prefer ``GEMINI_API_KEY`` (more specific), fall back to
        # ``GOOGLE_API_KEY`` for parity with the underlying provider adapter.
        key = (
            api_key
            or os.environ.get("GEMINI_API_KEY")
            or os.environ.get("GOOGLE_API_KEY")
        )
        if not key and not kwargs.get("vertexai"):
            raise ValueError(
                "Google Gemini LLM requires an api_key. Pass api_key='AIza...' or "
                "set GEMINI_API_KEY (or GOOGLE_API_KEY) in the environment."
            )
        super().__init__(api_key=key, model=model, **kwargs)

File: apps/patter-voice-sdk/libraries/typescript/src/llm/google.ts
/** Google Gemini LLM for Patter pipeline mode. */
import { GoogleLLMProvider as _GoogleLLM } from "../providers/google-llm";

/** Constructor options for the Google Gemini `LLM` adapter. */
export interface GoogleLLMOptions {
  /**
   * API key. Falls back to ``GEMINI_API_KEY`` first, then ``GOOGLE_API_KEY``.
   * (Google's CLI tooling uses ``GEMINI_API_KEY``; ``GOOGLE_API_KEY`` is the
   * legacy/alt name accepted for parity with other SDKs.)
   */
  apiKey?: string;
  /** Model id (e.g. ``"gemini-2.5-flash"``). */
  model?: string;
  /** Override the Generative Language API base URL (rarely needed). */
  baseUrl?: string;
  /** Sampling temperature. */
  temperature?: number;
  /** Maximum output tokens. */
  maxOutputTokens?: number;
}

/**
 * Google Gemini LLM provider (Developer API, streaming SSE).
 *
 * @example
 * ```ts
 * import * as google from "getpatter/llm/google";
 * const llm = new google.LLM();                                 // reads GEMINI_API_KEY or GOOGLE_API_KEY
 * const llm = new google.LLM({ apiKey: "AIza...", model: "gemini-2.5-flash" });
 * ```
 */
export class LLM extends _GoogleLLM {
  static readonly providerKey = "google";
  constructor(opts: GoogleLLMOptions = {}) {
    const key =
      opts.apiKey ?? process.env.GEMINI_API_KEY ?? process.env.GOOGLE_API_KEY;
    if (!key) {
      throw new Error(
        "Google LLM requires an apiKey. Pass { apiKey: 'AIza...' } or set " +
          "GEMINI_API_KEY (or GOOGLE_API_KEY).",
      );
    }
    super({
      apiKey: key,
      model: opts.model,
      baseUrl: opts.baseUrl,
      temperature: opts.temperature,
      maxOutputTokens: opts.maxOutputTokens,
    });
  }
}
